index.js

```diff
@@ -180,49 +206,57 @@ async function installLambdaFunction(lambdaArn, workDir, taskDir, layerDir) {
 * Starts heartbeat to indicate worker is working on the task
 *
 * @param {string} taskToken - the task token
-* @returns {intervalId} - interval id used by `clearInterval`
+* @param {integer} heartbeatInterval - number of milliseconds between heartbeat messages
+* @returns {SetIntervalAsyncTimer} - interval id used by `clearIntervalAsync`
 **/
-function startHeartbeat(taskToken) {
-  const sf = new AWS.StepFunctions({ apiVersion: '2016-11-23' });
-  return setInterval(() => {
-    sf.sendTaskHeartbeat({ taskToken }, (err) => {
-      if (err) {
-        log.error('error sending heartbeat', err);
-      }
+function startHeartbeat(taskToken, heartbeatInterval) {
```

Comment:
I had to update this code in order to migrate to aws-sdk v3 and its syntax. However, I'm pretty sure the previous code was wrong and wouldn't have worked anyway (e.g. the placement of the interval timer is wrong, right?). I think I've gotten it to what the original intent was but it wasn't tested and I'm guessing wasn't/isn't used by anyone. This code is skipped if the `heartbeat` option isn't passed to the CLI.

Reply:
It seems like we should add a test or tests to cover it? Or maybe take a quick look at the history to see if we can figure out where it originally came from. Definitely seems like what you have is right, would just like to understand if the original was broken and why it wasn't noticed.

Also could be a worth a quick look to see if we have any related Jira issues that someone reported? 

Basically, what's happening? 😆

Reply:
A git blame shows that block of code is 6 years old - so to be honest I'm not sure what the intent was since from day one I don't think it actually worked. I did a dive into JIRA and couldn't find anything related to this. Totally possible I missed something but nothing jumped out. By default this "feature" is disabled unless the user provides a `heartbeat` option to the CLI - so it's possible no one has actually used it.

You're right that we should have a test for it. I guess the question is if we should keep it at all since it was broken for years and never reported/requested. I can definitely add a test - `mockClient` should make that easy - to make sure the new code at least works even if we're unsure of the original intent.

Reply:
I definitely want to say "Add a ticket to remove/fix this" but I know that's unlikely to get prioritized. Removing it now feels a little scary on the off chance someone _is_ using it despite it being broken (not sure if it's completely busted or if there's still a side effect you could get out of it).

It's also not in scope here though. I think the only real option is to create a ticket and try to get it prioritized. Maybe we could add a comment here in code (if not already)? e.g. "TODO This doesn't appear to work."